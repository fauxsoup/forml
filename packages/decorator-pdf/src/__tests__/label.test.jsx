import Label from '../label';
import Context from '@forml/context';
import React from 'react';
import ReactPDF from '@react-pdf/renderer';
import { render } from '@testing-library/react';
import * as decorator from '../';

describe('renders', function () {
    let styles;
    let title;
    let form = { type: 'fieldset', items: [] };

    beforeEach(function () {
        title = jest.fn(() => ({ border: '1px solid red' }));
        styles = {
            get title() {
                return title();
            },
        };
    });

    test('without formStyles', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Label form={form} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });
    test('with relevant formStyles', function () {
        form = { type: 'checkbox', styles, key: [] };

        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Label form={form} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
        expect(title).toHaveBeenCalled();
    });
});
