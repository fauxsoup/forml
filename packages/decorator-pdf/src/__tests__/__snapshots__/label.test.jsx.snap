// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with relevant formStyles 1`] = `
<div>
  <page>
    <text
      style="font-size: 10pt; font-weight: bold; padding-top: 4pt; margin-bottom: 2pt; border: 1px solid red;"
    />
  </page>
</div>
`;

exports[`renders without formStyles 1`] = `
<div>
  <page>
    <text
      style="font-size: 10pt; font-weight: bold; padding-top: 4pt; margin-bottom: 2pt;"
    />
  </page>
</div>
`;
